rclcpp/test/rclcpp/test_timer.cpp

```diff
@@ -151,3 +152,47 @@ TEST_F(TestTimer, test_run_cancel_timer)
   EXPECT_TRUE(has_timer_run.load());
   EXPECT_TRUE(timer->is_canceled());
 }
+
+TEST_F(TestTimer, test_bad_arguments) {
+  auto callback = []() {};
+  auto node_base = rclcpp::node_interfaces::get_node_base_interface(test_node);
+  auto context = node_base->get_context();
+
+  auto steady_clock = std::make_shared<rclcpp::Clock>(RCL_STEADY_TIME);
+
+  // Negative period
+  EXPECT_THROW(
+    rclcpp::GenericTimer<decltype(callback)>(
+      steady_clock, -1ms, std::forward<decltype(callback)>(callback), context),
```

Comment:
@brawner why forwarding in this context? BTW the [use of `std::move` in `WallTimer` constructor on a `FunctorT&&` argument](https://github.com/ros2/rclcpp/blob/4f14a0cc142f43c62e40e5fadca2528183fc0a48/rclcpp/include/rclcpp/timer.hpp#L275) seems wrong and there I would've used forwarding.

Reply:
Switched std::forward to std::move, here and below.